python/tskit/genotypes.py

```diff
@@ -294,6 +295,45 @@ def frequencies(self, remove_missing=None) -> dict[str, float]:
             if not (allele is None and remove_missing)
         }
 
+    def __str__(self) -> str:
+        """
+        Return a plain text summary of the contents of a variant.
+        """
+        try:
+            self.site
```

Comment:
Is this line needed? I know the intent is to check the variant is decoded, but that happens a few lines lower anyway.